x-pack/legacy/plugins/apm/public/components/app/ServiceMap/icons.ts

```diff
@@ -52,10 +52,10 @@ const serviceIcons: { [key: string]: string } = {
 const isIE11 = !!window.MSInputMethodContext && !!document.documentMode;
 
 export function iconForNode(node: cytoscape.NodeSingular) {
-  const type = node.data('type');
+  const type = node.data('span.type');
 
-  if (type === 'service') {
-    return serviceIcons[node.data('agentName') as string];
+  if (node.data('service.name')) {
+    return serviceIcons[node.data('agent.name') as string];
```

Comment:
```suggestion
    return serviceIcons[node.data(SERVICE_AGENT_NAME) as string];
```

While you are at it, you might wanna rename `SERVICE_AGENT_NAME` to `AGENT_NAME` since it's no longer part of the service namespace.